src/components/profile/userContext.js

```diff
@@ -0,0 +1,388 @@
+
+  if (ctx === undefined) {
+    throw new Error('useUserListData must be used within a UserProvider');
+  }
+
+  const {
+    userList,
+    userError,
+    refreshUserList,
+    isLoadingUserList,
+    lastRefreshUserList,
+  } = ctx;
+
+  if (
+    !isLoadingUserList &&
+    (_.keys(userList).length === 0 ||
+      lastRefreshUserList === null ||
+      lastRefreshUserList >= Date.now + 600)
+  ) {
+    refreshUserList();
+  }
+  // What we want this consumer hook to actually return
+  return { refreshUserList, userList, userError, isLoadingUserList };
+};
+
+export const useUserData = () => {
+  const ctx = useContext(userContext);
+
+  if (ctx === undefined) {
+    throw new Error('useUserData must be used within a UserProvider');
+  }
+
+  const { user, userError, isLoadingUserData } = ctx;
+
+  return { user, userError, isLoadingUserData };
+};
+
+export const useCurrentUserData = () => {
+  const ctx = useContext(userContext);
+
+  if (ctx === undefined) {
+    throw new Error('useCurrentUserData must be used within a UserProvider');
+  }
+
+  const {
+    isLoadingUserData,
+    currentUser,
+    getCurrentUserData,
+    userError,
+    isAuthenticated,
+  } = ctx;
+
+  return {
+    isLoadingUserData,
+    currentUser,
+    getCurrentUserData,
+    isAuthenticated,
+    userError,
+  };
+};
+
+export const useEditUserDetailData = () => {
+  const ctx = useContext(userContext);
+
+  if (ctx === undefined) {
+    throw new Error('useEditUserDetailData must be used within a UserProvider');
+  }
+
+  const { isLoadingEditUserDetail, editUserDetail } = ctx;
+
+  return { isLoadingEditUserDetail, editUserDetail };
+};
+
+export const useUploadImageData = () => {
+  const ctx = useContext(userContext);
+
+  if (ctx === undefined) {
+    throw new Error('useUploadImageData must be within a UserProvider');
+  }
+
+  const { userError, uploadImage } = ctx;
+
+  return { userError, uploadImage };
+};
+
+export const useUserAuthenticationData = () => {
+  const ctx = useContext(userContext);
+
+  if (ctx === undefined) {
+    throw new Error(
+      'useUserAuthenticationData must be used within a UserProvider'
+    );
+  }
+
+  const {
+    isAuthenticated,
+    userError,
+    getAuthenticated,
+    isLoadingAuthenticated,
+  } = ctx;
+
+  if (!isLoadingAuthenticated && !isAuthenticated) {
+    getAuthenticated();
+  }
+
+  return {
+    getAuthenticated,
+    isAuthenticated,
+    userError,
+    isLoadingAuthenticated,
+  };
+};
+
+export const useUserLoginData = () => {
+  const ctx = useContext(userContext);
+
+  if (ctx === undefined) {
+    throw new Error('useUserLoginData must be used within a UserProvider');
+  }
+
+  const { loginUser, userError, isLoadingLogin } = ctx;
+
+  return { loginUser, userError, isLoadingLogin };
+};
+
+export const useUserRegisterData = () => {
+  const ctx = useContext(userContext);
+
+  if (ctx === undefined) {
+    throw new Error('useUserRegisterData must be used within a UserProvider');
```

Comment:
You _really_ should refactor this provider check out as a separate thing in a followup PR

Reply:
Will get to it when I'm sure all functionalities are working again, post Redux.